core/type_inference/infer.py

```diff
@@ -0,0 +1,70 @@
+import subprocess
+import argparse
+
+import hm_algo
+import jsonnet_ast as ast
+from lambda_types import TypeVariable
+from translate_jsonnet_to_lambda import translate_to_lambda_ast
+from rename import rename_local
+
+
+def get_jsonnet_ast_str(jsonnet, rebuild):
+    if rebuild:
+        command1 = ['bazel', 'build', '//core/type_inference:print_ast']
+        subprocess.run(command1, check=True, text=True)
+
+    command2 = ['bazel-bin/core/type_inference/print_ast', f'{jsonnet}']
+    result = subprocess.run(command2, stdout=subprocess.PIPE, 
+                            check=True, text=True, shell=False)
+    return result.stdout
+
+
+def parse_ast(ast_str):
+    return eval(ast_str)
+
+
+def create_init_env():
+    init_env = {}
+    init_env["__record_count__"] = 0
+    init_env["__plus_count__"] = 0
+    init_env["None"] = TypeVariable()
```

Comment:
Do we still need this? We needed it for a hack earlier on where we returned None for some translations. Are there still any of these left? We should probably replace them with Identifier "null" instead of None.

Reply:
Thanks for pointing me out on that. No, we don't need it anymore, nodes are either translated or exception is thrown.